_posts/2021-12-14-strimzi-and-log4shell.md

```diff
@@ -0,0 +1,248 @@
+* Kafka Exporter
+* TLS sidecars
+* Kaniko container builder
+
+Let's have a look at these components and identify those which are and are not affected by the Log4j2 vulnerability.
+
+### Unaffected components
+
+Several of the components are not affected because they do not use Java at all.
+This includes the [Kafka Exporter](https://github.com/danielqsj/kafka_exporter) and [Strimzi Canary](https://github.com/strimzi/strimzi-canary) which are written in Golang.
+The [Kaniko builder](https://github.com/GoogleContainerTools/kaniko) used to build the Kafka Connect images is also written in Golang
+The TLS sidecars used in the Cruise Control and Entity Operator are based on [Stunnel](https://www.stunnel.org/) which is written in C.
+
+The last unaffected components is the [Strimzi Drain Cleaner](https://github.com/strimzi/drain-cleaner).
+The Drain Cleaner is written in Java, but it uses Log4j2 library only in tests and not when used by the users.
+The next release of Drain Cleaner will contain the Log4j2 fix (for these tests) as well.
+But until then, you do not need to be worried if you use Drain Cleaner in your environment.
+
+### Affected components
+
+That leaves the five affected components.
+All Strimzi operators up-to (including) version 0.26.0 use Log4j2.
+So does the Strimzi Kafka Bridge up-to (including) version 0.20.3.
+The versions of [Cruise Control](https://github.com/linkedin/cruise-control) used by Strimzi operators from 0.22.0 up until 0.26.0 are affected as well.
+
+The operators and Cruise Control are normally accessible only internally within your Kubernetes cluster.
+We are currently not aware of any way of exploiting the vulnerability, either through Apache Kafka or directly.
+But we cannot exclude such possibility.
+
+The Strimzi Kafka Bridge is used to access the Kafka cluster using HTTP or AMQP protocols.
+As such, it is often exposed to more open networks and environments which makes it vulnerable.
+
+### What about Kafka?
+
+[Apache Kafka project](https://kafka.apache.org/) is currently using the [Log4j 1 library](https://logging.apache.org/log4j/1.2/).
+This is a predecessor of Log4j2 and is not affected by CVE-2021-44228.
+This includes not just the Kafka brokers, but also ZooKeeper, Kafka Connect and Mirror Maker 1 & 2.
+
+While in this case using Log4j 1 proved useful, its usage has its own problems.
+Log4j 1 is no longer maintained and has its own CVEs which are not as critical as CVE-2021-44228.
+The Apache Kafka project is working on replacing it with Log4j2 in the future.
+
+One of the CVEs in Log4j 1 is [CVE-2021-4104](https://nvd.nist.gov/vuln/detail/CVE-2021-4104) which is very similar to the Log4Shell CVE.
+But there is one main difference - it affects Log4j 1 only when you use the `JMSAppender` with some specific configuration.
+The `JMSAppender` is not used by default.
+So make sure you did not enabled it in your Kafka or ZooKeeper logging configuration and you should not be affected by it.
```

Comment:
Is there a log4j properties files in Strimzi to check when appenders have been enabled?

Reply:
It depends how you configured the logging. Normally, to configure the JMSAppender, the user would have it in some ConfigMap which they created.